### DISCUSSION TITLE
ffmpeg Vulkan Video API/decoder  and CUDA

### DISCUSSION THREAD

**Discussion by bhack** [PyAV-Org/PyAV#1276] [1. Help] [answered]
Is it possible to use [FFMPEG Vulkan decoder](https://trac.ffmpeg.org/wiki/HWAccelIntro#Vulkan) and then [access with cuda](https://github.com/FFmpeg/FFmpeg/blob/master/libavutil/hwcontext_vulkan.c#L3572) using PyAV?

**Comment by WyattBlue** [comment, marked as answer]
Currently, no. Historically we haven't supported any `hwaccel`s. I think we will add hwaccel support in the future, but it will probably take until v13.0.0 (October 2024) or v14.0.0 (2025) until we could ship such a feature.

**Comment by bhack** [comment]
We had some PR time ago like https://github.com/PyAV-Org/PyAV/pull/331.
I think the vulkan advantage is that it is internal so we don't need third party dependency:
https://trac.ffmpeg.org/wiki/HWAccelIntro
